File: Python/Easy/217.py
from typing import List
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_dict = {}
        for i in nums:
            if i in num_dict:
                num_dict[i] += 1
                if num_dict[i] >= 2:
                    return True
            else:
                num_dict[i] = 1
        return False

File: C++/Easy/217.cc
#include <vector>
#include <unordered_map>
#include <unordered_set>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (int i : nums){
            // find return iterator
            if (seen.count(i)){
                return true;
            } else{
                seen.insert(i);
            }
        }
        return false;
    }
};